api/web3server.go

```diff
@@ -311,11 +312,36 @@ func (svr *web3Handler) getBlockByNumber(in *gjson.Result) (interface{}, error)
 	return svr.getBlockWithTransactions(blk.Block, blk.Receipts, isDetailed.Bool())
 }
 
+func (svr *web3Handler) checkInputBlockNumber(str string) error {
```

Comment:
according to the eth standards, this `str` could be either a block number, a block enum, or a block hash.

Reply:
and prefer name it as `checkInputBlock`

Reply:
https://ethereum.org/en/developers/docs/apis/json-rpc/#default-block

> When requests are made that act on the state of Ethereum, the last default block parameter determines the height of the block.

> The following options are possible for the defaultBlock parameter:

> HEX String - an integer block number
> String "earliest" for the earliest/genesis block
> String "latest" - for the latest mined block
> String "safe" - for the latest safe head block
> String "finalized" - for the latest finalized block
> String "pending" - for the pending state/transactions

Reply:
should see this  https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getbalance